team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							let moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGen && (dexEntry.tier.indexOf('Bank') >= 0 || moveGen !== learnedGen.toString() || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
New plan lets us use `noPastGenBreeding` here.

`noPastGenBreeding` guarantees that `learnedGen` is `7`, and that `noPastGen` is true, which should be used to simplify the code here.